src/extensions/default/RemoteFileAdapter/main.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2018 - present Adobe Systems Incorporated. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var AppInit         = brackets.getModule("utils/AppInit"),
+        FileSystem      = brackets.getModule("filesystem/FileSystem"),
+        QuickOpen       = brackets.getModule("search/QuickOpen"),
+        PathUtils       = brackets.getModule("thirdparty/path-utils/path-utils"),
+        CommandManager  = brackets.getModule("command/CommandManager"),
+        Commands        = brackets.getModule("command/Commands"),
+        RemoteFile      = require("RemoteFile");
+
+    var HTTP_PROTOCOL = "http:",
+        HTTPS_PROTOCOL = "https:";
+
+    AppInit.htmlReady(function () {
+        var protocolAdapter = {
```

Comment:
Since this is an extension shouldn't the definition of ProtocolAdapter be provided by Brackets-Core for consistency.

Reply:
@vickramdhawal Sorry for the confusion. `protocolAdapter` is just a variable to hold the custom imprint of the adapter definition to be passed as protocol adapters for `HTTP:` and `HTTPS:`. Registration happens by invoking this part -
 ```javascript
// Register the custom object as HTTP and HTTPS protocol adapter
FileSystem.registerProtocolAdapter(HTTP_PROTOCOL, protocolAdapter);
FileSystem.registerProtocolAdapter(HTTPS_PROTOCOL, protocolAdapter);
```
I can't really define an interface (which if was possible, would have been part of core) and `Duck Typing` would be a gimmick as type checking is not something what we are interested about here.